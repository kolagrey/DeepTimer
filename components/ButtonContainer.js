import React from 'react';
import { View } from 'react-native';
import styles from '../styles';

const ButtonContainer = (props) => {
    return (
        <View style={styles.controlButtonContainer}>
            {props.children}
        </View>
    );
};

export default ButtonContainer;