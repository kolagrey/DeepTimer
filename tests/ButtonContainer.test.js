import React from 'react';
import { Text } from 'react-native';
import renderer from 'react-test-renderer';

import ButtonContainer from '../components/ButtonContainer';


describe('<ButtonContainer/>', () => {

    test('renders correctly', () => {
        const tree = renderer.create(<ButtonContainer/>).toJSON();
        expect(tree).toMatchSnapshot();
    });

    it("is of type 'View'", () => {
        const tree = renderer
            .create(
                <ButtonContainer />,
            ).toJSON();
        expect(tree.type).toBe('View');
    });

    it('can hold a child', () => {
        const tree = renderer
            .create(
                <ButtonContainer>
                    <Text>
                        Hello Test
                    </Text>
                </ButtonContainer>,
            ).toJSON();
        expect(tree.children.length).toBe(1);
    });
});