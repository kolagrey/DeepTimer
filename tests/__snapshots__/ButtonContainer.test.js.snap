// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ButtonContainer/> renders correctly 1`] = `
<View
  style={
    Object {
      "flex": 1,
      "flexDirection": "row",
    }
  }
/>
`;
